src/test/java/core/basesyntax/model/FruitTransactionTest.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class FruitTransactionTest {
+
+    @Test
+    void testFruitTransactionConstructorAndGetters() {
+        FruitTransaction.Operation operation = FruitTransaction.Operation.SUPPLY;
+        String fruit = "Apple";
+        int quantity = 10;
+
+        FruitTransaction transaction = new FruitTransaction(operation, fruit, quantity);
+
+        assertEquals(operation, transaction.getOperation());
+        assertEquals(fruit, transaction.getFruit());
+        assertEquals(quantity, transaction.getQuantity());
+    }
+
+    @Test
+    void testSetters() {
+        FruitTransaction transaction = new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                "Banana", 5);
+
+        transaction.setOperation(FruitTransaction.Operation.RETURN);
+        transaction.setFruit("Orange");
+        transaction.setQuantity(15);
+
+        assertEquals(FruitTransaction.Operation.RETURN, transaction.getOperation());
+        assertEquals("Orange", transaction.getFruit());
+        assertEquals(15, transaction.getQuantity());
+    }
+
+    @Test
+    void testOperationFromCodeValid() {
+        String code = "s";
+
+        FruitTransaction.Operation operation = FruitTransaction.Operation.fromCode(code);
+
+        assertEquals(FruitTransaction.Operation.SUPPLY, operation);
+    }
+
+    @Test
+    void testOperationFromCodeInvalid() {
```

Comment:
add valid test case. 
Check all method names. Use pattern `method_state_expectedResult`for example: `fromCode_invalidCode_notOk`